packages/dbgpt-core/src/dbgpt/util/speech/macos_tts.py

```
@@ -14,9 +14,12 @@ def _setup(self) -> None:
         if voice_index == 0:
-            os.system(f'say "{text}"')
+            args = ["say"]
         elif voice_index == 1:
-            os.system(f'say -v "Ava (Premium)" "{text}"')
+            args = ["say", "-v", "Ava (Premium)"]
         else:
-            os.system(f'say -v Samantha "{text}"')
+            args = ["say", "-v", "Samantha"]
+        # Pass the text as a separate argument (no shell) so it cannot be
+        # interpreted as a shell command, regardless of its content.
+        subprocess.run([*args, text], check=False)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)